app/controllers/SubmittedReturnsController.scala

```diff
@@ -101,7 +101,7 @@ class SubmittedReturnsController @Inject()(mcc: MessagesControllerComponents,
 
         case result =>
 
-          val obligations = result.flatMap(_.toSeq).flatMap(_.obligations)
+          val obligations = result.flatMap(_.toSeq).flatMap(_.obligations).distinct
```

Comment:
I think you have added this to the wrong controller as if I go to view http://localhost:9151/vat-through-software/vat-returns/return-deadlines then I can still see the duplicate returns being displayed